integration/test/Testlib/Env.hs

```diff
@@ -134,6 +147,8 @@ configName = \case
   Spar -> "spar"
   BackgroundWorker -> "background-worker"
   Stern -> "stern"
+  FederatorInternal -> "federator"
+  FederatorExternal -> "federator"
```

Comment:
```suggestion
  FederatorInternal -> "federator-internal"
  FederatorExternal -> "federator-external"
```

how did you test this?  it shouldn't have worked.

btw, took me too long to figure this out: `Testlib.Run` is looking in `/services/integration.yaml` for configuration options, and we already have both internal and external federator there.

no idea why this function is called `configName`, since the config data keys services config under (the camel-cased) `serviceName`.

Reply:
Changing this to have `-internal` and `-external` for the `configName` leads to the following errors.
Notably this line `[/home/owen/wire/wire-server/services/federator-external/federator-external.integration.yaml](run-services: Error while parsing /home/owen/wire/wire-server/services/federator-external/federator-external.integration.yaml: YAML exception:)`

Leaving the configName as just `federator` for both allows `make cr` to spin up services as expected. Once the services were running, I used `make devTest` to run the new tests.

```
owen@calcifer ~/wire/wire-server (git)-[WPB-3266] % make cr
./hack/bin/cabal-fmt.sh all
cabal build  all || ( make clean-hint; false )
Up to date
./hack/bin/cabal-install-artefacts.sh all
./dist/brig-schema --keyspace brig_test --replication-factor 1 > /dev/null
./dist/galley-schema --keyspace galley_test --replication-factor 1 > /dev/null
./dist/gundeck-schema --keyspace gundeck_test --replication-factor 1 > /dev/null
./dist/spar-schema --keyspace spar_test --replication-factor 1 > /dev/null
./dist/brig-schema --keyspace brig_test2 --replication-factor 1 > /dev/null
./dist/galley-schema --keyspace galley_test2 --replication-factor 1 > /dev/null
./dist/gundeck-schema --keyspace gundeck_test2 --replication-factor 1 > /dev/null
./dist/spar-schema --keyspace spar_test2 --replication-factor 1 > /dev/null
./integration/scripts/integration-dynamic-backends-db-schemas.sh --replication-factor 1 > /dev/null
./dist/brig-index reset --elasticsearch-index-prefix directory --elasticsearch-server http://localhost:9200 > /dev/null
./dist/brig-index reset --elasticsearch-index-prefix directory2 --elasticsearch-server http://localhost:9200 > /dev/null
./integration/scripts/integration-dynamic-backends-brig-index.sh --elasticsearch-server http://localhost:9200 > /dev/null
./services/run-services
[cannon@b.example.com] cannon: Network.Socket.bind: resource busy (Address already in use)
[cannon@example.com] cannon: Network.Socket.bind: resource busy (Address already in use)
[federator@b.example.com] federator: Network.Socket.bind: resource busy (Address already in use)
[federator@example.com] federator: Network.Socket.bind: resource busy (Address already in use)
run-services: Error while parsing /home/owen/wire/wire-server/services/federator-external/federator-external.integration.yaml: YAML exception:
Yaml file not found: /home/owen/wire/wire-server/services/federator-external/federator-external.integration.yaml
make: *** [Makefile:119: cr] Error 1
```

Some further digging however shows that this is actually a problem with the derived `Bounded` instance for `Service`. Somewhere in the service config override code that `make cr` calls it does use the `configName` function to build the path to the service directory to find a yaml file. Changing the Bounded instance to what it was prior to this new test allows `make cr` to spin up successfully.
```
instance Bounded Service where
  minBound = Brig
  maxBound = Stern
```
Luckily the integration.yaml and the federator.integration.yaml match for host and ports. However, this does make federator somewhat special in it's config handling for local dev environments which may lead to confusing test results. A more prudent method may be to change the `serviceHostPort` to allow for services with more than one HostPort..

Reply:
Done, but I also removed FederatorInternal and FederatorExternal from the Bounded instance range, for the above reasons.